src/api/firebase.js

```diff
@@ -70,11 +74,62 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
+/**
+ * Update an item in the user's list in Firestore.
+ * @param {string} listId The id of the list that has the items we're updating.
+ * @param {string} itemID The id of the list item we're updating.
+ * @param {boolean} checked The check state from ListItem.jsx.
+ */
+export async function updateItem(listId, itemId, checked) {
+	const listItemRef = doc(db, listId, itemId);
+	const listItemSnap = await getDoc(listItemRef);
+
+	const currentTotalPurchases = listItemSnap.data().totalPurchases;
+	let totalPurchases = currentTotalPurchases;
+
```

Comment:
Appreciate the readability of this function. I'll have to spend some more time to really understand how this Firebase function is operating but I appreciate how easy to read it all is.